pkg/secrethub/configdir/dir.go

```diff
@@ -0,0 +1,71 @@
+package configdir
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/mitchellh/go-homedir"
+)
+
+type Dir struct {
+	Path string
```

Comment:
Should this be publicly adjustable?

Reply:
i.e. should it be exported?

Reply:
> Should this be publicly adjustable?

Yes that is explicitly a feature. When the CLI is passed the `--config-dir` flag, you need to be able to create a config dir with a given path. 

Whether that should be with a `New(path string)` function, `Set(path string)` function or just my modifying the field is a style choice.

Reply:
Ahh, from the current `Dir` type, I expected a one-on-one mapping between a directory on your file system and a struct, but if I understand you correctly, the struct only contains the configuration of which directory to use and can be switched to point to a different directory?

To prevent similar confusion, could we change the name to reflect this? e.g. `Config`

Alternatively, we could use the `New(path string)` to create a new `Dir` instance when using a different directory.

Reply:
Yes, the types in the `configdir.Dir` and `configdir.CredentialFile` merely point to a location on the filesystem and offer utility methods for working with those files/dirs. 

To me the name `configdir.Dir` makes sense, it's the directory where configuration is stored. Later we may add a `configdir.ConfigFile` or `configdir.SettingsFile` with configuration. The `configdir.XXXFile` types are stored within the `configdir.Dir` type. 

Now the `New(path string)` thingy could actually work. May even be a bit nicer because you're reducing the surface area of contact of the package. Not suuuper necessary now, but a nice improvement I'd say.

@jpcoenen your call what to do.

Reply:
@SimonBarendse: have a look at how I solved this.